src/iterators/mod.rs

```diff
@@ -1050,22 +1050,22 @@ clone_bounds!(
     AxisChunksIter['a, A, D] {
         @copy {
             life,
-            last_ptr,
+            original_len,
         }
         iter,
         last_dim,
     }
 );
 
 fn chunk_iter_parts<A, D: Dimension>(v: ArrayView<A, D>, axis: Axis, size: usize)
-    -> (AxisIterCore<A, D>, *mut A, D)
+    -> (AxisIterCore<A, D>, usize, D)
 {
     let axis = axis.index();
     let axis_len = v.shape()[axis];
     let size = if size > axis_len { axis_len } else { size };
     let last_index = axis_len / size;
```

Comment:
I realize this isn't part of the PR, but `last_index` shouldn't really be named `last_index`, because it's the last index only in the non-evenly-divisible case. (Consider e.g. `axis_len == 3` and `size == 1` -> `last_index == 3`.) I'd consider inlining the division in the computation of `iter_len` since `last_index` isn't used anywhere else.)